### PR TITLE
Resurrection python3

### PR DESCRIPTION
**Pull request by janwijbrand**
Python 3 compatibility fixes.

Note how the doubt expressed in the `setindex.txt` tests about the validity of heterogeneous values in the index proofed to be justified: heterogeneous might have implicitly worked in Python 2, they won't in Python 3. Since the doubt about this was so clearly expressed, I took the decision to make this explicit: values in the (set)index should be of the same type for the BTrees to apply ordering.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -5,8 +5,9 @@ CHANGES
 1.6.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
-
+- Python3 compatibily. Note how this implies the indexes can only store values
```
Comment (mgedmin) [inline CHANGES.txt]: I suggest "Python 3" with a space.

File: CHANGES.txt [hunk 2]
```diff
@@ -5,8 +5,9 @@ CHANGES
...
+  of the same type from now on. The possible necessity for this was already
+  expressed in the tests for setindex and is now explicit.
 
 1.6 (2013-07-04)
```
Comment (mgedmin) [inline CHANGES.txt:13]: Two blank lines before version sections, for consistency.

**Review by mgedmin — APPROVED** [review]
LGTM!  :fireworks:

File: CHANGES.txt (hunk 1 above)
Comment (janwijbrand) [reply CHANGES.txt]: We'll do this.

File: CHANGES.txt (hunk 2 above)
Comment (janwijbrand) [reply CHANGES.txt:13]: We'll do this.

**Comment by jamadden** [comment]
This was handled in #3.

**Pull request closed without merging**